src/code/z_lights.c

```diff
@@ -27,11 +95,106 @@ void Lights_PointGlowSetInfo(LightInfo* info, s16 x, s16 y, s16 z, u8 r, u8 g, u
     Lights_PointSetInfo(info, x, y, z, r, g, b, radius, LIGHT_POINT_GLOW);
 }
 
+/**
+ * POINT_LIGHT_R0 (called r0 in some calculations below) is used to normalize a light radius r when converting it to
+ * attenuation coefficients to send to the microcode.
+ *
+ * The RSP accepts 3 attenuation coefficients controlling a point light:
+ *  - The quadratic attenuation kq
+ *  - The quadratic attenuation kl
+ *  - The quadratic attenuation kc
+ *
+ * The attenuation computes the light intensity at a vertex as a function of distance from the source:
+ *  I(d) = I / ((kq / 524288) * d * d + (kl / 32768) * d + (kc / 16) + (1 / 2))
+ * where
+ *  - I is the source intensity, based on the light color
+ *  - d is the distance from the light position to the vertex
+ *
+ * In this implementation we let kc = 8 always, this is the smallest allowed value as overflows will occur for kc < 8
+ * where the ucode would compute a reciprocal > 1.0.
+ *
+ * We let kl = (r0/r) and kq = SQ(r0/r) where r is the light radius parameter that various functions accept.
+ *
+ * If we let POINT_LIGHT_R0=3000, the attenuation I(d) will have the characteristics
+ *      I(r/4) ~ 0.5I
+ *      I(r/2) ~ 0.2I
+ *      I(r)   ~ 0.06I
```

Comment:
Probably should also mention this is only valid for r (the light radius) in a certain range (which I forgot but is documented somewhere in the escaperoom repo's z_lights)